src/main/scala/examples/Adder.scala

```diff
@@ -29,7 +29,7 @@ class Adder(val n:Int) extends Module {
     FAs(i).b := io.B(i)
     FAs(i).cin := carry(i)
     carry(i+1) := FAs(i).cout
-    sum(i) := FAs(i).sum.toBool()
+    sum(i) := FAs(i).sum.asBool()
```

Comment:
What's the rule for `.asBool` vs `.asBool()` again?

Reply:
In principle it should be `asBool` since it doesn't modify the object, but this potentially runs afoul of `apply` chaining (or implicits?). I think @ducky64 can explain it better than I can.

Reply:
I'm not sure what the recommendation was, if any. No parens seems logical if it works. I can't think off the top of my head why the parens would affect implicit chaining.